include/simdjson/dom/array.h

```diff
@@ -23,23 +23,42 @@ class array {
 
   class iterator {
   public:
+    using value_type = element;
+    using difference_type = std::ptrdiff_t;
+
     /**
      * Get the actual value
      */
-    inline element operator*() const noexcept;
+    inline value_type operator*() const noexcept;
```

Comment:
How does this show up on doxygen though? Are we introducing a layer of indirection before people can understand what's going on?

Minor problem even if so; I don't really care that much about the documentation on the iterator interface methods, when iteration doesn't generally expose those methods to the user.

Reply:
I am not sure, but I assume it will use the alias and add a level of indirection. This matches other sources of documentation, for example, notice the return type of `size()` here: https://en.cppreference.com/w/cpp/container/vector/size